src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -263,7 +263,9 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 : context.SemanticModel.LookupSymbols(context.LeftToken.SpanStart);
 
             // Filter out any extension methods that might be imported by a using static directive.
-            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod());
+            // But include extension methods declared in the context's type
+            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod()
+            || symbol.ContainingType.DeclaringSyntaxReferences.Any(x => x.GetSyntax().IsEquivalentTo(context.ContainingTypeDeclaration))).ToList();
```

Comment:
This line needs, at the very least, to be indented.  I'd also prefer it be pulled out into a helper method.